### PR TITLE
Permission for pip install #35

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem
- User is not allowed to install PIP packages. #35 

## Solution
- *Set permission to .venv folder*

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [ ] **Code review** : 
- [ ] **Approval (LGTM)** : 
- [ ] **Approval (LGTM)** : 
- **CI succeeded** : 
[Build Status](https://ci-apps-hq.yunohost.org/jenkins/job/jupyterlab_ynh%20PR41/)  
*Please replace '-NUM-' in this link by the PR number.*  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by kay0u** [comment]
Hi, tested and working! Thank you very much.

But there still remain a problem. During the upgrade, pipenv will erase the old env, and create a fresh one with only these [packages](https://github.com/ericgaspar/jupyterlab_ynh/blob/d442160f55a5d1adbc8b25343be730c6b6ed3584/scripts/install#L102) and not the packages installed via pip.

You can test it with this command:
```
YNH_FORCE_UPGRADE=1 yunohost app upgrade jupyterlab -u https://github.com/ericgaspar/jupyterlab_ynh/tree/permission-for-pip-install-%2335
```

I don't know if we want this behaviour and I don't know how to fix this (if we consider this as a problem?..)

File: scripts/upgrade [hunk 1]
```diff
@@ -116,7 +116,7 @@ then
 
 	pushd $final_path
 
-	PIPENV_VENV_IN_PROJECT="enabled" PIPENV_SKIP_LOCK=true  ynh_exec_warn_less python3 -m pipenv install jupyterlab==$jupyterlab_version jupyterhub notebook jupyterhub-ldapauthenticator --three
```
Comment (ericgaspar) [inline scripts/upgrade:119]: The `-- three` argument seems to be destructive and may not be needed in the upgrade script.

**Comment by ericgaspar** [comment]
I still have these warnings after updating JupiterLab from the master branch...

```
> Warning: Group 'visitors' already has permission 'jupyterlab.main' enabled
> Warning: This permission is currently granted to all users in addition to other groups. You probably want to either remove the 'all_users' permission or remove the other groups it is currently granted to.
> Warning: The permission was not updated because the addition/removal requests already match the current state.
```

File: scripts/upgrade (hunk 1 above)
Comment (kay0u) [reply scripts/upgrade:119]: Good catch! Tested and working!

File: scripts/install [hunk 2]
```diff
@@ -160,8 +160,8 @@ ynh_store_file_checksum --file="$final_path/config/jupyter_notebook_config.py"
 #=================================================
 
 # Set permissions to app files
-chown -R root: $final_path
-chown -R $admin: $final_path/.venv/share
```
Comment (kay0u) [inline scripts/install:164]: In fact, if we only change `/share' with `/lib' it works too, I'd prefer to restrict permissions as much as possible.


On the other hand, if a normal user (not the admin) tries to install a new package with pip, he will have the same problem as before, are you aware of that? It's not necessarily a problem to let only the admin install a new package, what do you think?

Comment (ericgaspar) [reply scripts/install:164]: You are right, It works. I guess it makes sens to restrict the install of packages to admin for now.

**Pull request merged**